SignalMessaging/environment/OWSAudioSession.swift

```diff
@@ -38,8 +38,10 @@ public class OWSAudioSession: NSObject {
     public func startAmbientAudioActivity(_ audioActivity: AudioActivity) {
         Logger.debug("\(logTag) in \(#function)")
 
-        startAudioActivity(audioActivity)
+        objc_sync_enter(self)
```

Comment:
I wonder if we should introduce a `SyncLock`, e.g. 
```
let _ = SyncLock(self)
```
`SyncLock` would be RIAA - take a lock when initialized and release the lock when deallocated.  With let, the lock should be retained throughout the enclosing block.  This is fine too, TIOLI